### PR TITLE
v3: Enforce key length for EncryptCookie middleware default functions

### PR DESCRIPTION
**Pull request by gaby**
## Description

- Enforce key length for default encryptor/decryptor in `EncryptCookie` middleware.
  - `AES-GCM` only works with `16`, `24`, `32` bytes keys.
  - This will allow the user to use AES-128 if speed is critical.
- Add `length` param to `encryptcookie.GenerateKey`.
- Add unit-test for `encryptcookie.GenerateKey`.
- Add benchmarks for `EncryptCookie` middleware.
- Add parallel benchmarks for `EncryptCookie` middleware.
- Increase Middleware Test Coverage by `+12.37%`

Related to #3048 

## Changes introduced

List the new features or adjustments introduced in this pull request. Provide details on benchmarks, documentation updates, changelog entries, and if applicable, the migration guide.

- [x] Benchmarks: Describe any performance benchmarks and improvements related to the changes.
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [x] Changelog/What's New: Include a summary of the additions for the upcoming release notes.

## Type of change

- [x] Enhancement (improvement to existing features and functionality)
- [x] Documentation update (changes to documentation)

### CODE REVIEW

File: middleware/encryptcookie/utils.go [hunk 1]
```diff
@@ -17,6 +17,11 @@
 		return "", fmt.Errorf("failed to base64-decode key: %w", err)
 	}
 
+	keyLen := len(keyDecoded)
+	if keyLen != 16 && keyLen != 24 && keyLen != 32 {
+		return "", errors.New("encryption key must be 16, 24, or 32 bytes")
+	}
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/encryptcookie/utils.go:23]: Done

File: middleware/encryptcookie/utils.go [hunk 2]
```diff
@@ -43,6 +47,12 @@
 	if err != nil {
 		return "", fmt.Errorf("failed to base64-decode key: %w", err)
 	}
+
+	keyLen := len(keyDecoded)
+	if keyLen != 16 && keyLen != 24 && keyLen != 32 {
+		return "", errors.New("encryption key must be 16, 24, or 32 bytes")
+	}
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/encryptcookie/utils.go:54]: Done

File: middleware/encryptcookie/utils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/encryptcookie/utils.go:23]: Yes they have been added for both encrypt and decrypt

**Review by efectn — APPROVED** [review]

File: docs/middleware/encryptcookie.md [hunk 3]
```diff
@@ -16,8 +16,8 @@ This middleware encrypts cookie values and not the cookie names.
 // Intitializes the middleware
 func New(config ...Config) fiber.Handler
 
-// Returns a random 32 character long string
-func GenerateKey() string
+// Returns a random 16, 24, 32 bytes encoded string
+func GenerateKey(length) string
```
Comment (sixcolors) [inline docs/middleware/encryptcookie.md:20]:
```suggestion
// GenerateKey returns a random string of 16, 24, or 32 bytes.
// The length of the key determines the AES encryption algorithm used:
// 16 bytes for AES-128, 24 bytes for AES-192, and 32 bytes for AES-256-GCM.
func GenerateKey(length) string
```

File: middleware/encryptcookie/config.go [hunk 4]
```diff
@@ -18,18 +18,19 @@ type Config struct {
...
 	// Custom function to encrypt cookies.
 	//
-	// Optional. Default: EncryptCookie
+	// Optional. Default: EncryptCookie (using AES-256-GCM)
 	Encryptor func(decryptedString, key string) (string, error)
 
 	// Custom function to decrypt cookies.
 	//
-	// Optional. Default: DecryptCookie
+	// Optional. Default: DecryptCookie (using AES-256-GCM)
```
Comment (sixcolors) [inline middleware/encryptcookie/config.go:33]: Without knowing the input key length we don't really have a default algo anymore do we?

**Review by sixcolors — DISMISSED** [review]

File: middleware/encryptcookie/config.go (hunk 4 above)
Comment (gaby) [reply middleware/encryptcookie/config.go:33]: I can remove `256`, its AES-GCM by default

File: middleware/encryptcookie/utils.go [hunk 5]
```diff
@@ -10,13 +10,20 @@
...
 func EncryptCookie(value, key string) (string, error) {
 	keyDecoded, err := base64.StdEncoding.DecodeString(key)
 	if err != nil {
 		return "", fmt.Errorf("failed to base64-decode key: %w", err)
 	}
 
+	keyLen := len(keyDecoded)
+	if keyLen != 16 && keyLen != 24 && keyLen != 32 {
+		return "", ErrInvalidKeyLength
+	}
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/encryptcookie/utils.go:25]: It's already there

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**